ext/standard/crc32.c

```diff
@@ -116,9 +104,51 @@ PHP_FUNCTION(crc32)
 	p += nr_simd;
 #endif
 
+	/* The trailing part */
 	for (; nr--; ++p) {
 		crc = ((crc >> 8) & 0x00FFFFFF) ^ crc32tab[(crc ^ (*p)) & 0xFF ];
 	}
+
+	return crc;
+}
+
+int crc32_stream_bulk_update(uint32_t *crc, php_stream *fp, size_t nr)
+{
+	size_t handled = 0, n;
+	char buf[1024];
+
+	while (handled < nr) {
+		n = nr - handled;
+		n = (n < sizeof(buf)) ? n : sizeof(buf); /* tweak to buf size */
+
+		n = php_stream_read(fp, buf, n);
+		if (n > 0) {
+			*crc = crc32_bulk_update(*crc, buf, n);
+			handled += n;
+		} else { /* EOF */
+			return FAILURE;
```

Comment:
NIT `EOF` might be not directly an error, since it loops and `EOL` might be delivered not in the first iteration. Probably ok if it's only about local files, but stream can be anything (also remote). Perhaps it's better to return `handled` or `EOF`.  Anyway this does the job and we don't export.

Thanks!